pkg/cloud/vsphere/provisioner/govmomi/create.go

```diff
@@ -4,49 +4,115 @@ import (
+const (
+	// localIPV4lookup resolves via cloudinit and looks up the instance's IP through the provider's metadata service.
+	// See https://cloudinit.readthedocs.io/en/latest/topics/instancedata.html
+	localIPV4Lookup = "{{ ds.meta_data.local_ipv4 }}"
+
+	// hostnameLookup resolves via cloud init and uses cloud provider's metadata service to lookup its own hostname.
+	hostnameLookup = "{{ ds.meta_data.hostname }}"
+
+	// containerdSocket is the path to containerd socket.
+	containerdSocket = "/var/run/containerd/containerd.sock"
+
+	// nodeRole is the label assigned to every node in the cluster.
+	nodeRole = "node-role.kubernetes.io/node="
+)
+
+func (pv *Provisioner) Create(
+	ctx context.Context,
+	cluster *clusterv1.Cluster,
+	machine *clusterv1.Machine,
+	bootstrapToken string) error {
+
 	if cluster == nil {
 		return errors.New(constants.ClusterIsNullErr)
 	}
 
-	klog.V(4).Infof("govmomi.Actuator.Create %s", machine.Name)
+	machineRole := vsphereutils.GetMachineRole(machine)
+	if machineRole == "" {
+		return errors.Errorf(
+			"Unable to get machine role while creating machine with GoVmomi "+
+				"%s=%s %s=%s %s=%s %s=%s",
+			"cluster-namespace", cluster.Namespace,
+			"cluster-name", cluster.Namespace,
+			"machine-namespace", machine.Namespace,
+			"machine-name", machine.Name)
+	}
+
+	klog.V(4).Infof("Creating machine with GoVmomi %s=%s %s=%s %s=%s %s=%s %s=%s",
+		"cluster-namespace", cluster.Namespace,
+		"cluster-name", cluster.Namespace,
+		"machine-namespace", machine.Namespace,
+		"machine-name", machine.Name,
+		"machine-role", machineRole)
+
 	s, err := pv.sessionFromProviderConfig(cluster, machine)
 	if err != nil {
-		return err
+		return errors.Wrapf(
+			err,
+			"Unable to get session while creating machine with GoVmomi "+
+				"%s=%s %s=%s %s=%s %s=%s",
+			"cluster-namespace", cluster.Namespace,
+			"cluster-name", cluster.Namespace,
+			"machine-namespace", machine.Namespace,
+			"machine-name", machine.Name)
 	}
+
 	createctx, cancel := context.WithCancel(*s.context)
 	defer cancel()
-	usersession, err := s.session.SessionManager.UserSession(createctx)
+
+	userSession, err := s.session.SessionManager.UserSession(createctx)
 	if err != nil {
-		return err
-	}
-	klog.V(4).Infof("Using session %v", usersession)
+		return errors.Wrapf(
+			err,
+			"Unable to get user session while creating machine with GoVmomi "+
+				"%s=%s %s=%s %s=%s %s=%s",
+			"cluster-namespace", cluster.Namespace,
+			"cluster-name", cluster.Namespace,
+			"machine-namespace", machine.Namespace,
+			"machine-name", machine.Name)
+	}
+
+	klog.V(4).Infof("Got user session while creating machine with GoVmomi "+
+		"%s=%s %s=%s %s=%s %s=%s %s=%s %s=%v",
+		"cluster-namespace", cluster.Namespace,
+		"cluster-name", cluster.Namespace,
+		"machine-namespace", machine.Namespace,
+		"machine-name", machine.Name,
+		"machine-role", machineRole,
+		"user-session", userSession)
```

Comment:
Haven't looked at the entire userSession struct, but is it safe to log this? Also how big is this struct? Might make logs unreadable if you dump the entire struct like this.

Reply:
It was logged before, so I'm still logging it. We can double-check it though.